src/core_modules/capture-core/components/WidgetStagesAndEvents/Stages/Stage/StageDetail/hooks/helpers.ts

```diff
@@ -41,12 +41,12 @@ const getEventStatus = (event: ApiEnrollmentEvent) => {
     return { status: event.status, options: undefined };
 };
 
-const getValueByKeyFromEvent = (event: ApiEnrollmentEvent, { id, resolveValue }: Object) => {
+const getValueByKeyFromEvent = (event: ApiEnrollmentEvent, { id, resolveValue }: Record<string, unknown>) => {
     if (resolveValue) {
-        return resolveValue(event);
+        return (resolveValue as (eventParam: ApiEnrollmentEvent) => unknown)(event);
     }
 
-    return event[id];
+    return (event as Record<string, unknown>)[id as string];
 };
```

Comment:
1) Add`type FieldKey = { id: string; resolveValue?: (event: ApiEnrollmentEvent) => any }; `
2) Change to` { id, resolveValue }: FieldKey`
3) Rollback to `return resolveValue(event);` and `return event[id];`